src/app/components/media/MediaTimeline.js

```diff
@@ -0,0 +1,469 @@
+        };
+      }
+
+      entities[`clip-${name}`].instances.push({
+        id,
+        start_seconds,
+        end_seconds,
+        url: new URL(`#t=${start_seconds},${end_seconds}&id=${encodeURIComponent(id)}`, document.location.href).href,
+      });
+    });
+
+    tags
+      .filter(({ node }) => !!node)
+      .forEach(({
+        node,
+        node: {
+          id: instance,
+          fragment,
+          parsed_fragment: { t: [start_seconds, end_seconds] = [] } = {},
+          tag_text_object,
+        },
+      }) => {
+        if (!tag_text_object) return;
+
+        const { id, text: name } = tag_text_object;
+        if (!fragment) {
+          projecttags.push({ id, name });
+          return;
+        }
+
+        if (!entities[id]) {
+          entities[id] = {
+            id, name, project_tag: { id, name }, instances: [], node,
+          };
+        }
+
+        entities[id].type = 'tag';
+        entities[id].instances.push({ start_seconds, end_seconds, id: instance });
+      });
+
+    locations.filter(({ node: { id } }) => !!id).forEach(({
+      node, node: { id, parsed_fragment: { t: [start_seconds, end_seconds] }, content },
+    }) => {
+      const {
+        geolocation_viewport: { viewport, zoom = 10 },
+        geolocation_location,
+      } = JSON.parse(content).reduce((acc, { field_name, value_json }) =>
+        ({ ...acc, [field_name]: value_json }), {});
+
+      const {
+        properties: { name = id.trim() } = {},
+        geometry: { type = 'Point', coordinates = [0, 0] } = {},
+      } = geolocation_location;
+      const [lng, lat] = coordinates;
+
+      if (!entities[`place-${name}`]) {
+        entities[`place-${name}`] = {
+          id: `place-${name}`,
+          name,
+          viewport,
+          zoom,
+          lat,
+          lng,
+          type: type === 'Point' ? 'marker' : 'polygon',
+          test: { geolocation_location },
+          project_place: { id: `place-${name}`, name },
+          instances: [],
+          node,
+        };
+
+        if (type === 'Polygon') {
+          entities[`place-${name}`].polygon = coordinates[0].map(([lng, lat]) => ({ lat, lng }));
+        }
+      }
+
+      entities[`place-${name}`].instances.push({ id, start_seconds, end_seconds });
+    });
+
+    const data = {
+      project: {
+        projectclips: projecttags,
+        projecttags,
+        projectplaces: projecttags,
+      },
+      commentThreads,
+      videoClips: Object.values(entities).filter(({ project_clip }) => !!project_clip),
+      videoTags: Object.values(entities).filter(({ project_tag }) => !!project_tag),
+      videoPlaces: Object.values(entities).filter(({ project_place }) => !!project_place),
+      user: {
+        first_name,
+        id: userId,
+        last_name: '',
+        profile_img_url,
+      },
+    };
+
+    console.log({ data });
+
+    return {
+      duration, time, data, mediaId, dbid,
+    };
+  }
+
+  componentDidUpdate() {
```

Comment:
the initial render has practically no data, I can have it in there too but it won't do anything

Reply:
I'm not understanding: why does the initial render have practically no data? (I've been expecting the data to come from GraphQL, since this component has no state -- only props.)

Reply:
fixed by removing this until the functionality is made in check-ui